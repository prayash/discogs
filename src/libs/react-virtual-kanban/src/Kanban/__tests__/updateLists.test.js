import { updateLists } from '../updateLists';

const lists = [
  {
    id: 1,
    rows: [
      {id: 1},
      {id: 2}
    ]
  },
  {
    id: 2,
    rows: [
      {id: 3},
      {id: 4}
    ]
  }
];

test('move lists', () => {
  const updatedList = updateLists(lists, {
    from: {listIndex: 0},
    to: {listIndex: 1}
  });

  expect(JSON.stringify(updatedList)).toMatchSnapshot();
});

test('move item inside same list', () => {
  const updatedList = updateLists(lists, {
    from: {listIndex: 0, rowIndex: 0},
    to: {listIndex: 0, rowIndex: 1}
  });

  expect(JSON.stringify(updatedList)).toMatchSnapshot();
});

test('move between lists', () => {
  const updatedList = updateLists(lists, {
    from: {listIndex: 0, rowIndex: 1},
    to: {listIndex: 1, rowIndex: 0}
  });

  expect(JSON.stringify(updatedList)).toMatchSnapshot();
});

test('move item to an empty list', () => {
  const otherLists = [
    {
      id: 1,
      rows: [
        {id: 1},
        {id: 2}
      ]
    },
    {
      id: 2,
      rows: []
    }
  ];

  const updatedList = updateLists(otherLists, {
    from: {listIndex: 0, rowIndex: 0},
    to: {listIndex: 1, rowIndex: 0}
  });

  expect(JSON.stringify(updatedList)).toMatchSnapshot();
});

test('move item from a list with a single element', () => {
  const otherLists = [
    {
      id: 1,
      rows: [
        {id: 1},
        {id: 2}
      ]
    },
    {
      id: 2,
      rows: [
        {id: 3}
      ]
    }
  ];

  const updatedList = updateLists(otherLists, {
    from: {listIndex: 1, rowIndex: 0},
    to: {listIndex: 0, rowIndex: 2}
  });

  expect(JSON.stringify(updatedList)).toMatchSnapshot();
});

test('null move', () => {
  const updatedList = updateLists(lists, {
    from: {listIndex: 0, rowIndex: 0},
    to: {listIndex: 0, rowIndex: 0}
  });

  expect(updatedList).toMatchObject(lists);
});

test('lists immutability', () => {
  const updatedList = updateLists(lists, {
    from: {listIndex: 0, rowIndex: 0},
    to: {listIndex: 0, rowIndex: 1}
  });

  expect(updatedList).not.toBe(lists);
});

test('single list equality', () => {
  const updatedList = updateLists(lists, {
    from: {listIndex: 0},
    to: {listIndex: 1}
  });

  expect(updatedList[1]).toBe(lists[0]);
});

test('single item equality', () => {
  const updatedList = updateLists(lists, {
    from: {listIndex: 0, rowIndex: 0},
    to: {listIndex: 0, rowIndex: 1}
  });

  expect(updatedList[0][1]).toBe(lists[0][0]);
});
