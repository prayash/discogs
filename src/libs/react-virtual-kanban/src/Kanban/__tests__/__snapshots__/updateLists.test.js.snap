exports[`test move between lists 1`] = `"[{\"id\":1,\"rows\":[{\"id\":1}]},{\"id\":2,\"rows\":[{\"id\":2},{\"id\":3},{\"id\":4}]}]"`;

exports[`test move item from a list with a single element 1`] = `"[{\"id\":1,\"rows\":[{\"id\":1},{\"id\":2},{\"id\":3}]},{\"id\":2,\"rows\":[]}]"`;

exports[`test move item inside same list 1`] = `"[{\"id\":1,\"rows\":[{\"id\":2},{\"id\":1}]},{\"id\":2,\"rows\":[{\"id\":3},{\"id\":4}]}]"`;

exports[`test move item to an empty list 1`] = `"[{\"id\":1,\"rows\":[{\"id\":2}]},{\"id\":2,\"rows\":[{\"id\":1}]}]"`;

exports[`test move lists 1`] = `"[{\"id\":2,\"rows\":[{\"id\":3},{\"id\":4}]},{\"id\":1,\"rows\":[{\"id\":1},{\"id\":2}]}]"`;
